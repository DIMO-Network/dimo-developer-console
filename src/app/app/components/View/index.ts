import View from './View';

export { View };

export default View;
