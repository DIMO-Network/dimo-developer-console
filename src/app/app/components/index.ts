export * from './OnboardingCard';
export * from './Onboarding';
