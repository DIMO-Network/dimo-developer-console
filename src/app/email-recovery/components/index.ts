export * from './EmailRecoveryForm';
export * from './CheckEmail';