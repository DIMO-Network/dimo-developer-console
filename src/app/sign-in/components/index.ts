export * from './OtpInputForm';
export * from './PasskeyLogin';
export * from './SignInMethodForm';
