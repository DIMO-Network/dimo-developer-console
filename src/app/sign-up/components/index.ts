export * from './BuildForForm';
export * from './CompanyInfoForm';
export * from './SignUpWith';
export * from './UserInfoForm';
