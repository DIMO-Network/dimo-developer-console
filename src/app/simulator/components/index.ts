export { View } from './View';
export { Header } from './Header';
export { SimulatedVehiclesSection } from './SimulatedVehiclesSection';
export { CreateSimulatedVehicleModal } from './CreateSimulatedVehicleModal';
