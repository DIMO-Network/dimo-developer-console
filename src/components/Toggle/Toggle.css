.custom-toggle {
  @apply relative inline-flex h-5 w-10 flex-shrink-0 cursor-pointer items-center justify-center rounded-full;

  .base {
    @apply pointer-events-none absolute w-full rounded-md bg-white;
  }

  .bar {
    @apply pointer-events-none absolute mx-auto h-4 w-9 rounded-full transition-colors duration-200 ease-in-out;

    &.active {
      @apply bg-primary-400;
    }

    &.inactive {
      @apply bg-dark-grey-300;
    }
  }

  .dot {
    @apply pointer-events-none absolute left-0 inline-block h-5 w-5 transform rounded-full border border-gray-200 shadow ring-0 transition-transform duration-200 ease-in-out;

    &.active {
      @apply translate-x-5 bg-primary-300;
    }

    &.inactive {
      @apply translate-x-0 bg-dark-grey-400;
    }
  }
}
