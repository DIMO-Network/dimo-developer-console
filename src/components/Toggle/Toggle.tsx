import { useState, type FC } from 'react';
import { Switch } from '@headlessui/react';
import classNames from 'classnames';

import './Toggle.css';

interface IProps {
  checked?: boolean;
  onToggle?: (v: boolean) => void;
}

export const Toggle: FC<IProps> = ({
  checked = false,
  onToggle = () => {},
}) => {
  const [enabled, setEnabled] = useState(checked);

  const handleToggle = (value: boolean) => {
    setEnabled(value);
    onToggle(value);
  };
  return (
    <Switch
      checked={enabled}
      onChange={handleToggle}
      className="custom-toggle group"
      aria-label="toggle"
    >
      <span className="sr-only">Use setting</span>
      <span
        aria-hidden="true"
        className="base"
      />
      <span
        aria-hidden="true"
        className={classNames({ active: enabled, inactive: !enabled }, 'bar')}
      />
      <span
        aria-hidden="true"
        className={classNames({ active: enabled, inactive: !enabled }, 'dot')}
      />
    </Switch>
  );
};

export default Toggle;
