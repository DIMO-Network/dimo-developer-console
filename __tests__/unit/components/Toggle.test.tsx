import { fireEvent, render, screen } from '@testing-library/react';

import { Toggle } from '@/components/Toggle';

describe('Toggle', () => {
  it('renders a toggle component', () => {
    const handleChange = jest.fn();

    const { container } = render(<Toggle onToggle={handleChange} />);

    const toggleElm = screen.getByLabelText('toggle');

    expect(container).toMatchSnapshot();
    expect(toggleElm).toBeInTheDocument();

    fireEvent.click(toggleElm);

    expect(handleChange).toHaveBeenCalledTimes(1);
    expect(handleChange).toHaveBeenCalledWith(true);

    fireEvent.click(toggleElm);

    expect(handleChange).toHaveBeenCalledWith(false);
  });
});
