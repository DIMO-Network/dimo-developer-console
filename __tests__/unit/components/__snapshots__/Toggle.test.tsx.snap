// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders a toggle component 1`] = `
<div>
  <button
    aria-checked="false"
    aria-label="toggle"
    class="custom-toggle group"
    data-headlessui-state=""
    id="headlessui-switch-:r0:"
    role="switch"
    tabindex="0"
    type="button"
  >
    <span
      class="sr-only"
    >
      Use setting
    </span>
    <span
      aria-hidden="true"
      class="base"
    />
    <span
      aria-hidden="true"
      class="inactive bar"
    />
    <span
      aria-hidden="true"
      class="inactive dot"
    />
  </button>
</div>
`;
